### PR TITLE
extend pkcs7 openssl bindings

### PR DESCRIPTION
**Pull request by ddcc**

### CODE REVIEW

**Comment by alex** [comment]
Jenkins, ok to test.

File: src/_cffi_src/openssl/pkcs7.py [hunk 1]
```diff
@@ -44,13 +77,17 @@
 PKCS7 *PKCS7_encrypt(Cryptography_STACK_OF_X509 *, BIO *,
                      const EVP_CIPHER *, int);
 int PKCS7_decrypt(PKCS7 *, EVP_PKEY *, X509 *, BIO *, int);
+
+BIO* PKCS7_dataInit (PKCS7 *, BIO *);
```
Comment (reaperhulk) [inline src/_cffi_src/openssl/pkcs7.py]: Our style is to do `BIO *method` rather than `BIO* method`

File: src/_cffi_src/openssl/pkcs7.py [hunk 2]
```diff
@@ -10,7 +10,40 @@
...
+
+typedef ... PKCS7_DIGEST;
+typedef ... PKCS7_ENCRYPT;
+typedef ... PKCS7_ENVELOPE;
+
+typedef struct pkcs7_st {
     ASN1_OBJECT *type;
+    union {
+        char *ptr;
+        /* NID_pkcs7_data */
```
Comment (reaperhulk) [inline src/_cffi_src/openssl/pkcs7.py]: Please remove the comments (both here and up in the PKCS7_SIGN_ENVELOPE struct)

File: src/_cffi_src/openssl/pkcs7.py [hunk 3]
```diff
@@ -10,7 +10,40 @@
 
 TYPES = """
 typedef struct {
+    Cryptography_STACK_OF_X509 *cert;       /* [ 0 ] */
+    Cryptography_STACK_OF_X509_CRL *crl;    /* [ 1 ] */
```
Comment (alex) [inline src/_cffi_src/openssl/pkcs7.py]: You can drop these comments, they don't explain anything.

File: src/_cffi_src/openssl/pkcs7.py [hunk 4]
```diff
@@ -10,7 +10,40 @@
...
+    ...;
+} PKCS7_SIGN_ENVELOPE;
+
+typedef ... PKCS7_DIGEST;
+typedef ... PKCS7_ENCRYPT;
+typedef ... PKCS7_ENVELOPE;
+
+typedef struct pkcs7_st {
```
Comment (alex) [inline src/_cffi_src/openssl/pkcs7.py]: You can drop the `pkcs7_st`.

File: src/_cffi_src/openssl/pkcs7.py (hunk 1 above)
Comment (alex) [inline src/_cffi_src/openssl/pkcs7.py]: Small style nits: the `*` should be next to `PKCS7`, not `BIO`, and no space after `Init`

File: src/_cffi_src/openssl/pkcs7.py (hunk 3 above)
Comment (ddcc) [inline src/_cffi_src/openssl/pkcs7.py]: Ok, I saw some comments in x509v3.py, so I wasn't sure whether to leave them in or not.

**Comment by reaperhulk** [comment]
Looks good, now to see what the CI makes of it :)

**Pull request merged**